lib/llm/src/http/service/openai.rs

```diff
@@ -70,29 +70,46 @@ pub type ErrorResponse = (StatusCode, Json<ErrorMessage>);
 
 #[derive(Serialize, Deserialize, Debug)]
 pub(crate) struct ErrorMessage {
-    error: String,
+    message: String,
+    #[serde(rename = "type")]
+    error_type: String,
+    code: u16,
+}
+
+fn map_error_code_to_error_type(code: u16) -> String {
+    match code {
+        404 => "NotFoundError".to_string(),
+        503 => "ServiceUnavailableError".to_string(),
+        500 => "InternalServerError".to_string(),
+        501 => "NotImplementedError".to_string(),
+        422 => "UnprocessableEntityError".to_string(),
+        400 => "BadRequestError".to_string(),
+        _ => "UnknownError".to_string(),
+    }
 }
 
 impl ErrorMessage {
     /// Not Found Error
     pub fn model_not_found() -> ErrorResponse {
-        (
-            StatusCode::NOT_FOUND,
-            Json(ErrorMessage {
-                error: "Model not found".to_string(),
-            }),
-        )
+        let code = StatusCode::NOT_FOUND;
+        let error_type = map_error_code_to_error_type(code.as_u16());
+        (code, Json(ErrorMessage {
+            message: "Model not found".to_string(),
+            error_type,
+            code: code.as_u16(),
+        }))
```

Comment:
If we make `message` be `canonical_reason()`, and `error_type` be something more generic (e.g. `client` for 4XX and `server` for 5XX), we can collapse all these methods into one.